### PR TITLE
LA-236 Preview file in Received Share

### PR DESCRIPTION
**Pull request by dab246**
Pls review this guys.

https://user-images.githubusercontent.com/80730648/112971834-ceb57180-9179-11eb-8c55-b735bd1a8396.mp4

### CODE REVIEW

File: lib/presentation/widget/received/received_share_viewmodel.dart [hunk 1]
```diff
@@ -173,6 +182,66 @@ class ReceivedShareViewModel extends BaseViewModel {
...
+    }
+  }
+
+  OnlineThunkAction _handleDownloadPreviewReceivedShare(ReceivedShare receivedShare, DownloadPreviewType downloadPreviewType, CancelToken cancelToken) {
+    return OnlineThunkAction((Store<AppState> store) async {
+      await _downloadPreviewReceivedShareInteractor
+          .execute(receivedShare, downloadPreviewType, cancelToken)
+          .then((result) => result.fold(
+              (failure) {
+            if (failure is DownloadPreviewReceivedShareFailure && !(failure.downloadPreviewException is CancelDownloadFileException)) {
```
Comment (manhduc-nguyen) [inline lib/presentation/widget/received/received_share_viewmodel.dart:220]: change your indent pls

File: lib/presentation/widget/received/received_share_viewmodel.dart [hunk 2]
```diff
@@ -173,6 +182,66 @@ class ReceivedShareViewModel extends BaseViewModel {
...
+              store.dispatch(ReceivedShareAction(Left(NoReceivedSharePreviewAvailable())));
+            }
+          },
+              (success) {
+            if (success is DownloadPreviewReceivedShareViewState) {
```
Comment (manhduc-nguyen) [inline lib/presentation/widget/received/received_share_viewmodel.dart:225]: same

**Review by manhduc-nguyen — APPROVED** [review]
LGTM

File: data/lib/src/datasource_impl/received_share_datasource_imp.dart [hunk 3]
```diff
@@ -91,4 +94,27 @@ class ReceivedShareDataSourceImp extends ReceivedShareDataSource {
...
+  @override
+  Future<Uri> downloadPreviewReceivedShare(ReceivedShare receivedShare,
+      DownloadPreviewType downloadPreviewType, Token permanentToken, Uri baseUrl, CancelToken cancelToken) {
+    var downloadUrl;
+    if (downloadPreviewType == DownloadPreviewType.original) {
+      downloadUrl = Endpoint.receivedShares
+          .downloadServicePath(receivedShare.shareId.uuid)
+          .generateDownloadUrl(baseUrl);
+    } else {
+      downloadUrl = Endpoint.receivedShares
```
Comment (manhduc-nguyen) [inline data/lib/src/datasource_impl/received_share_datasource_imp.dart:107]: oh, have you tested the download url for this case ?

Comment (dab246) [reply data/lib/src/datasource_impl/received_share_datasource_imp.dart:107]: Yes, i tested it

**Review by hoangdat — APPROVED** [review]

**Pull request merged**